src/connectedk8s/azext_connectedk8s/tests/latest/test_connectedk8s_scenario.py

```diff
@@ -5,98 +5,521 @@
+        self.cmd('aks get-credentials -g {rg} -n {managed_cluster_name} -f {kubeconfig}')
+        self.cmd('connectedk8s connect -g {rg} -n {name} -l eastus --tags foo=doo --kube-config {kubeconfig} --kube-context {managed_cluster_name}', checks=[
+            self.check('tags.foo', 'doo'),
+            self.check('name', '{name}')
+        ])
+        self.cmd('connectedk8s show -g {rg} -n {name}', checks=[
+            self.check('name', '{name}'),
+            self.check('resourceGroup', '{rg}'),
+            self.check('tags.foo', 'doo')
+        ])
+
+        self.cmd('aks create -g {rg} -n {managed_cluster_name_second} --generate-ssh-keys')
+        self.cmd('aks get-credentials -g {rg} -n {managed_cluster_name_second} -f {kubeconfigpls}')
+        self.cmd('connectedk8s connect -g {rg} -n {name_second} -l eastus --tags foo=doo --kube-config {kubeconfigpls} --kube-context {managed_cluster_name_second}', checks=[
+            self.check('tags.foo', 'doo'),
+            self.check('name', '{name_second}')
+        ])
+        self.cmd('connectedk8s show -g {rg} -n {name_second}', checks=[
+            self.check('name', '{name_second}'),
+            self.check('resourceGroup', '{rg}'),
+            self.check('tags.foo', 'doo')
         ])
-        self.cmd('connectedk8s delete -g akkeshar -n cliplscc --kube-config {kubeconfigpls} -y')
 
+        clusters_list = self.cmd('az connectedk8s list -g {rg}').get_output_in_json()
+        # fetching names of all clusters
+        cluster_name_list=[]
+        for clusterdesc in clusters_list:
+            cluster_name_list.append(clusterdesc['name'])
+
+        assert(len(cluster_name_list) == len(managed_cluster_list))
+
+        # checking if the output is correct with original list of cluster names
+        cluster_name_list.sort()
+        for i in range(0,len(cluster_name_list)):
+            assert(cluster_name_list[i] == managed_cluster_list[i])
+
+        # deleting the clusters
+        self.cmd('connectedk8s delete -g {rg} -n {name_second} --kube-config {kubeconfigpls} --kube-context {managed_cluster_name_second} -y')
+        self.cmd('aks delete -g {rg} -n {managed_cluster_name_second} -y')
+
+        self.cmd('connectedk8s delete -g {rg} -n {name} --kube-config {kubeconfig} --kube-context {managed_cluster_name} -y')
+        self.cmd('aks delete -g {rg} -n {managed_cluster_name} -y')
+
+        # delete the kube config
+        os.remove("%s" % (_get_test_data_file(managed_cluster_name + '-config.yaml')))
         os.remove("%s" % (_get_test_data_file('pls-config.yaml')))
 
-    def test_forcedelete(self):
 
-        managed_cluster_name = self.create_random_name(prefix='test-force-delete', length=24)
+    @live_only()
+    @ResourceGroupPreparer(name_prefix='conk8stest', location='eastus2euap', random_name_length=16)
+    def test_upgrade(self,resource_group):
+
+        managed_cluster_name = self.create_random_name(prefix='test-upgrade', length=24)
```

Comment:
Is it mandatory to create the aks cluster and onboard it every time? Is there any way with which we can onboard once and run other commands/tests on top of it . If not all, atleast if we can reuse it for some tests so that we can save time for the overall test run

Reply:
AKS cluster creation everytime will take a lot of time. What is the avg time our current pipeline is taking?

Reply:
all the tests actually run parallelly, the avg time is 50 mins but the point here is when the pipeline is triggered it just not only goes to our extension it sequentially goes to all of the extension spend around 2 mins each (again this is parallel too) and then executes our tests too. Pipeline ex - https://dev.azure.com/azclitools/internal/_build/results?buildId=33353&view=results

Having only one aks means , testing everything inside one function , this might actually increase the execution time of the tests and not sure if this is a cleaner approach

Reply:
Are all the tests run parallely or sequentially? If parallel, then it makes sense to create everytime. But if sequential, then we should reuse

Reply:
why does it go to other extensions if we set the pipeline variable to run only our extension ?

Reply:
Also, do we know how many tests run in parallel at once? Even with parallel if it is taking 50 mins, we should get that clarity

Reply:
I mean, how many tests are run in parallel for a particular extension

Reply:
thats how the pipeline is designed , will confirm with zelin but i think 8 can run parallely , connectedk8s execution time is around 37 mins

Reply:
Okay

Reply:
when we set the pipeline paramter , there is a parameter called user_parallelism = 8 by default , and I have seen 8 or maybe more extensions that are run parallely

Reply:
That is number of extensions run parallely, I am asking about number of tests per extension parallely.

Reply:
If tests per extension don't run parallely, then there would be no point in create aks and onboarding it everytime. Also, does the pipeline provide an option to run selected tests per extension or all tests will be run always

Reply:
The live test was not designed to be run every time. It will only run automatically at specific times, such as every Thursday morning.
Each extension will assign an agent to run the test, which is why you will see so many extensions running at the same time, but they all run for only two minutes, because they are skipped because the running conditions are not met.
If you want to run the tests on every commit, please remove the @live_only decorator and we will run the recording tests after every commit. (Please remember to upload the recording file generated by the live test)